### PR TITLE
Fix issue#82

### PR DESCRIPTION
**Pull request by Karthik2007**
#82 Fix issue

Change:
Reused KotlinDataClassFileGenerator() and changed the flow to create the list of non duplicate classes for inclass JsonToKotlin Conversion

### CODE REVIEW

**Comment by wuseal** [comment]
@Karthik2007 Glad to receive your PR, I have reviewed the commits, Some suggestions here:
1. Please not try to reuse `KotlinDataClassFileGenerator`  If possible, Because of it only designed for generating kotlin data class files, And this fix may not need to have a relationship with `PsiDirectory`, I have seen that you want to rename the duplicate class name from the files name in the same directory, but the name of kotlin files may not be the kotlin class name inside kotlin file. Also, the action class `MakeKotlinClassAction` could insert generated data classes inside any block not only the kotlin file top domain but also inside the functions or inner classes. so rename the kotlin class name depend on kotlin file should not be reliable.

1. If you want to implement what you want, I think the `PSI` system may be the comfortable solution. If it is ok, we will appreciate seeing your works on `PSI` aspect contributions.

1. Look at the issue of auto check system upon, resolve the relevant issue is also a good job for our project. :smile:

**Comment by Karthik2007** [comment]
@wuseal Thanks for reviewing the PR and providing the suggestions.

I will look into other way of doing it. Went ahead with reusing, since `KotlinDataClassFileGenerator` had the logic to rename the classes if already present. Will be checking the PSI system and will update the PR.

Happy Coding! :)

**Comment by wuseal** [comment]
@Karthik2007 Yes! Of course, we need to rename the duplicate class names in the same package when inserting into the top domain fo a kotlin file.
As class identifier like this `wu.seal.SomeClass`, then we could rename class name to new name according to the class identifier.
At first, we need to rename the duplicate class names in one conversion of one JSON string. Then we could try to rename logic deeper depends on all the classes in the project.
Now, this PR could be merged first after remove the resuing `KotlinDataClassFileGenerator` logic code. 
In the future, You could commit a new PR to resolve the class conflicts against the whole project classes to make rename logic perfect!
Happy Coding! 😃

**Comment by Karthik2007** [comment]
@wuseal Updated the PR. Removed the reuse of `KotlinDataClassFileGenerator` and PsiDirectory usage. I have used relevant code to `MakeKotlin` class action flow. 

Again the duplicate class name check from the same repository is pending. I will create separate issue for the same and work on that.

Please let me know if any changes needed.

Happy Coding!

**Comment by wuseal** [comment]
@Karthik2007 I am very happy that we can reach an agreement. And very thanks to your contributions on this issue, I will review your changes and reply to this PR tonight. 
About Duplicate class filter, I think you could have a look at the `Kotlin PSI` or `Java PSI` and I think it could solve the duplicate class  problem, Hope you will Breaking through new progress on this issue. Thanks 😄 
Happy Coding!

**Comment by wuseal** [comment]
@Karthik2007 Very happy it seems works well. only a little need to change that the notification is not comfortable for insert Class condition 
image

**Comment by Karthik2007** [comment]
@wuseal  oh Sorry, forgot to change that. 

will below change work?
 
Instead of _"Kotlin Data class files generated successful"_ - _"Kotlin Data Classes generated Successful"_

File: src/main/kotlin/wu/seal/jsontokotlin/KotlinDataClassMaker.kt [hunk 1]
```diff
@@ -1,21 +1,177 @@
...
+    }
+
+    // method to make in class data classes for JsonToKotlinClass, renames same class name to different
+    private fun makeKotlinDataClasses(
+            removeDuplicateClassCode: String) : List<ParsedKotlinDataClass>{
+
+        val kotlinClasses = generateKotlinDataClassesWithNonConflictNames(removeDuplicateClassCode = removeDuplicateClassCode)
+
+        val notifyMessage = buildString {
+            append("${kotlinClasses.size} Kotlin Data Class files generated successful")
```
Comment (wuseal) [inline src/main/kotlin/wu/seal/jsontokotlin/KotlinDataClassMaker.kt]:
```suggestion
            append("${kotlinClasses.size} Kotlin Data Classes generated successful")
```

Comment (Karthik2007) [reply src/main/kotlin/wu/seal/jsontokotlin/KotlinDataClassMaker.kt]: Resolved.

**Comment by wuseal** [comment]
@Karthik2007 Nice Job, And then you need to add a unit test for this issue to validate the case is working correctly, It's a workflow of this project to keep this issue will not happen again when next time modifies some code about this function.

**Comment by wuseal** [comment]
> @wuseal Can you please check and let me know why the ci is failing. Thanks.

I have trigger rebuild that CI, let's wait for the result again. 🙂

**Comment by wuseal** [comment]
@Karthik2007 I have checked the test, All works well.  Good job! Merged it

**Comment by Karthik2007** [comment]
Happy Coding ! :)

**Pull request merged**